src/validate.js

```diff
@@ -140,5 +140,15 @@ module.exports = validators => {
     return true;
   };
 
+  exports.throwVersionMismatchErrorIfType = (obj, classType) => {
+    if (is.object(obj) && obj.constructor.name === classType.name) {
```

Comment:
So if an admin object has been passed here, what would the two values be? `FieldValue` and `admin.firestore.FieldValue`?

Reply:
Tried this out: https://jsfiddle.net/cddxtq3a/1/. It's `FieldValue`.

Reply:
When would `obj.constructor.name` and `classType.name` would not be equal?

Reply:
Users can hit this codepath if they try to store custom objects that are initialized with new ...(). In our validator, we basically do the following:

```
if (instance FieldValue.delete) {
...
}
else if (instance FieldValue.timestamp) {
...
}
else if (!isPlainObject()) { // Check if object has a protoype
... throw 
}
```

By calling `throwVersionMismatchErrorIfType`, we can provide a better error for objects that share names with our types.